packages/block-editor/src/components/grid/grid-visualizer.js

```diff
@@ -118,19 +119,25 @@ const GridVisualizerGrid = forwardRef(
 function ManualGridVisualizer( { gridClientId, gridInfo } ) {
 	const [ highlightedRect, setHighlightedRect ] = useState( null );
 
-	const gridItems = useSelect(
-		( select ) => select( blockEditorStore ).getBlocks( gridClientId ),
+	const gridItemStyles = useSelect(
+		( select ) => {
+			const { getBlockOrder, getBlockStyles } = unlock(
+				select( blockEditorStore )
+			);
+			const blockOrder = getBlockOrder( gridClientId );
+			return getBlockStyles( blockOrder );
+		},
 		[ gridClientId ]
 	);
 	const occupiedRects = useMemo( () => {
 		const rects = [];
-		for ( const block of gridItems ) {
+		for ( const style of Object.values( gridItemStyles ) ) {
```

Comment:
I noticed that it means `occupiedRects` may no longer be in block order (as it's sourced from an object).

I think it's ok though, the only thing I can see it's used for is this code, which doesn't depend on the ordering:
```jsx
const isCellOccupied = occupiedRects.some( ( rect ) =>
	rect.contains( column, row )
);
```

Reply:
Yeah order isn't important.